core/model/modx/processors/system/dashboard/widget/feed.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * Class modDashboardWidgetFeedProcessor
+ *
+ * Used to load the news and security feeds on the dashboard over AJAX. The processed feed content
+ * (i.e. HTML) is returned in object->html.
+ */
+class modDashboardWidgetFeedProcessor extends modProcessor
+{
+    /**
+     * @var modRSSParser
+     */
+    protected $rss;
+
+    public function process()
+    {
+        $feed = $this->getProperty('feed', 'news');
+        if (!in_array($feed, array('news', 'security'), true)) {
+            return $this->failure('Invalid feed type');
+        }
+
+        $enabled = $this->modx->getOption('feed_modx_' . $feed . '_enabled', null, true);
+        if (!$enabled) {
+            return $this->failure();
+        }
+
+        $url = $this->modx->getOption('feed_modx_' . $feed);
+        // Check if the domain is valid before attempting to load it
+        $feedHost = parse_url($url, PHP_URL_HOST);
+        if ($feedHost && function_exists('checkdnsrr') && !checkdnsrr($feedHost, 'A')) {
+            return $this->failure();
+        }
+
+        return $this->loadFeed($url);
+    }
+
+    public function loadFeed($url)
+    {
+        $this->modx->loadClass('xmlrss.modRSSParser','',false,true);
```

Comment:
No particular reason. Mostly just copied the old code from the widgets into the processor, and that used it like this... getService is probably cleaner though.